include/zephyr/drivers/battery.h

```diff
@@ -0,0 +1,470 @@
+ */
+typedef int (*battery_is_cut_off)(const struct device *dev);
+
+/**
+ * Read battery vendor parameter.
+ *
+ * Vendor parameter handlers are implemented in a board-specific battery.c
+ *
+ * @param param		Parameter identifier.
+ * @param value		Location to store retrieved value.
+ * @return non-zero if error.
+ */
+typedef int (*battery_get_vendor_param)(const struct device *dev, uint32_t param, uint32_t *value);
+
+/**
+ * Write battery vendor parameter.
+ *
+ * Vendor parameter handlers are implemented in a board-specific battery.c
+ *
+ * @param param		Parameter identifier.
+ * @param value		Value to write to the battery.
+ * @return non-zero if error.
+ */
+typedef int (*battery_set_vendor_param)(const struct device *dev, uint32_t param, uint32_t value);
+
+/**
+ * Wait for battery stable.
+ *
+ * @return non-zero if error.
+ */
+typedef int (*battery_wait_for_stable)(const struct device *dev);
+
+/**
+ * Print all battery info for debugging purposes
+ */
+typedef void (*print_battery_debug)(const struct device *dev);
+
+/**
+ * Get the disconnect state of the battery.
+ */
+typedef enum battery_disconnect_state (*battery_get_disconnect_state)(const struct device *dev);
+
+/*
+ * If remaining charge is more than x% of the full capacity, the
+ * remaining charge is raised to the full capacity before it's
+ * reported to the rest of the system.
+ *
+ * Some batteries don't update full capacity timely or don't update it
+ * at all. On such systems, compensation is required to guarantee
+ * the remaining charge will be equal to the full capacity eventually.
+ *
+ * On some systems, Rohm charger generates audio noise when the battery
+ * is fully charged and AC is plugged. A workaround is to do charge-
+ * discharge cycles between 93 and 100%. On such systems, compensation
+ * was also applied to mask this cycle from users.
+ *
+ * We do it in the EC so it has more
+ * knowledge on the charger and the battery so it can perform more
+ * granular and precise compensation.
+ *
+ */
+typedef void (*battery_compensate_params)(const struct device *dev, struct batt_params *batt);
+
+/**
+ * board-specific battery_compensate_params
+ */
+typedef void (*board_battery_compensate_params)(const struct device *dev, struct batt_params *batt);
+typedef void (*battery_validate_params)(const struct device *dev, struct batt_params *batt);
+
+/**
+ * Read static battery info from a main battery and store it in a cache.
+ *
+ * @return EC_SUCCESS or EC_ERROR_*.
+ */
+typedef int (*update_static_battery_info)(const struct device *dev);
+
+/**
+ * Read dynamic battery info from a main battery and store it in a cache.
+ */
+typedef void (*update_dynamic_battery_info)(const struct device *dev);
+
+/**
+ * Return the board-specific battery type.
+ *
+ * @return a value of `enum battery_type`.
+ */
+typedef int (*board_get_battery_type)(const struct device *dev);
+
+/**
+ * Return 1 if CFET is disabled, 0 if enabled. -1 if an error was encountered.
```

Comment:
Don't use unexpanded abbreviations in help. What is CFET?  (Whether _I_ know is not the point).